src/sentry/notifications/helpers.py

```diff
@@ -461,6 +488,36 @@ def _get_subscription_values(
     return is_disabled, is_active, subscription
 
 
+def _get_subscription_values_v2(
+    group: Group,
+    subscriptions_by_group_id: Mapping[int, GroupSubscription],
+    notification_settings_by_provider: Mapping[ExternalProviders, NotificationSettingsOptionEnum],
+) -> tuple[bool, bool, GroupSubscription | None]:
+    is_disabled = False
+    subscription = subscriptions_by_group_id.get(group.id)
+    if subscription:
+        # Having a GroupSubscription overrides NotificationSettings.
+        is_active = subscription.is_active
+    else:
```

Comment:
it seems like we don't have any test coverage for this `else` block -- worth adding?